pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -134,3 +135,169 @@ func (pc *podConfigurator) reconcileMissingPods(ifConfigs []*interfacestore.Inte
+		return fmt.Errorf("failed to add Openflow entries for container %s: %v", containerID, err)
+	}
+
+	// Notify the Pod update event to required components.
+	event := types.PodUpdate{
+		PodName:      ifConfig.PodName,
+		PodNamespace: ifConfig.PodNamespace,
+		IsAdd:        true,
+		ContainerID:  ifConfig.ContainerID,
+	}
+	m.podUpdateNotifier.Notify(event)
+
+	// Remove the annotation from Pod if exists.
+	m.updatePodUnreadyAnnotation(ifConfig.PodNamespace, ifConfig.PodName, false)
+	return nil
+}
+
+func (m *podIfaceMonitor) updatePodUnreadyAnnotation(podNamespace, podName string, addAnnotation bool) {
+	pod, err := m.kubeClient.CoreV1().Pods(podNamespace).Get(context.Background(), podName, metav1.GetOptions{})
+	if err != nil {
+		klog.ErrorS(err, "Failed to get Pod when trying to update 'unready' annotations", "Namespace", podNamespace, "Name", podName)
+		return
+	}
+
+	annotated := false
+	if pod.Annotations != nil {
+		_, annotated = pod.Annotations[types.PodNotReadyAnnotationKey]
+	}
+
+	if addAnnotation && !annotated {
+		// Add the annotation on Pod with '"pod.antrea.io/not-ready": ""'
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]interface{}{types.PodNotReadyAnnotationKey: ""},
+			},
+		})
+		m.kubeClient.CoreV1().Pods(podNamespace).Patch(context.Background(), podName, apitypes.MergePatchType, patch, metav1.PatchOptions{})
+	} else if !addAnnotation && annotated {
+		// Remove the annotation on Pod with '"pod.antrea.io/not-ready": ""'
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]interface{}{types.PodNotReadyAnnotationKey: nil},
+			},
+		})
+		m.kubeClient.CoreV1().Pods(podNamespace).Patch(context.Background(), podName, apitypes.MergePatchType, patch, metav1.PatchOptions{})
+	}
+}
+
+func (m *podIfaceMonitor) updateUnReadyPod(status *openflow15.PortStatus) {
+	ovsPort := string(bytes.Trim(status.Desc.Name, "\x00"))
+	obj, found := m.unReadyInterfaces.Load(ovsPort)
+	if !found {
+		klog.InfoS("OVS port is not found", "ovsPort", ovsPort)
+		return
+	}
+	podInfo := obj.(*unReadyPodInfo)
+	ofPort := status.Desc.PortNo
+	if err := m.updatePodFlows(ovsPort, int32(ofPort)); err != nil {
+		klog.ErrorS(err, "Failed to update Pod's OpenFlow entries", "PodName", podInfo.podName, "PodNamespace", podInfo.podNamespace, "OVSPort", ovsPort)
+		return
+	}
+	// Delete the Pod from unReadyPods
+	m.unReadyInterfaces.Delete(ovsPort)
+}
+
+func (m *podIfaceMonitor) checkUnReadyPods() {
+	m.unReadyInterfaces.Range(func(key, value any) bool {
+		podInfo := value.(*unReadyPodInfo)
+		if !podInfo.annotated && time.Now().Sub(podInfo.createTime).Seconds() > podNotReadyTimeInSeconds {
+			m.updatePodUnreadyAnnotation(podInfo.podNamespace, podInfo.podName, true)
+			podInfo.annotated = true
```

Comment:
I think this should be conditional: patching the Pod can fail and we want to be able to try again at the next tick.